smartsim/_core/generation/generator.py

```diff
@@ -28,206 +28,276 @@
+        """Build and return a Jobs base directory. The path is created by combining the
+        root directory with the Job type (derived from the class name),
+        the name attribute of the Job, and an index to differentiate between multiple
+        Job runs.
 
-        :param job: The Job instance for which the root directory is generated.
-        :param job_index: The index of the Job instance (used for naming).
-        :returns: The path to the root directory for the Job instance.
+        :param job: The job object
+        :param job_index: The index of the Job
+        :returns: The built file path for the Job
         """
         job_type = f"{job.__class__.__name__.lower()}s"
         job_path = self.root / f"{job_type}/{job.name}-{job_index}"
         return pathlib.Path(job_path)
 
-    def _generate_run_path(self, job: Job, job_index: int) -> pathlib.Path:
-        """Generates the path for the "run" directory within the root directory
-        of a specific Job instance.
+    def _build_job_run_path(self, job: Job, job_index: int) -> pathlib.Path:
+        """Build and return a Jobs run directory. The path is formed by combining
+        the base directory with the `run` class-level variable, where run specifies
+        the name of the job's run folder.
 
-        :param job (Job): The Job instance for which the path is generated.
-        :param job_index (int): The index of the Job instance (used for naming).
-        :returns: The path to the "run" directory for the Job instance.
+        :param job: The job object
+        :param job_index: The index of the Job
+        :returns: The built file path for the Job run folder
         """
-        path = self._generate_job_root(job, job_index) / "run"
+        path = self._build_job_base_path(job, job_index) / self.run
         path.mkdir(exist_ok=False, parents=True)
         return pathlib.Path(path)
 
-    def _generate_log_path(self, job: Job, job_index: int) -> pathlib.Path:
-        """
-        Generates the path for the "log" directory within the root directory of a specific Job instance.
+    def _build_job_log_path(self, job: Job, job_index: int) -> pathlib.Path:
+        """Build and return a Jobs log directory. The path is formed by combining
+        the base directory with the `log` class-level variable, where log specifies
+        the name of the job's log folder.
 
-        :param job: The Job instance for which the path is generated.
-        :param job_index: The index of the Job instance (used for naming).
-        :returns: The path to the "log" directory for the Job instance.
+        :param job: The job object
+        :param job_index: The index of the Job
+        :returns: The built file path for the Job run folder
         """
-        path = self._generate_job_root(job, job_index) / "log"
+        path = self._build_job_base_path(job, job_index) / self.log
         path.mkdir(exist_ok=False, parents=True)
```

Comment:
Some of these ``_build`` functions call ``mkdir`` and some don't.  I woudl proposed that the ``_build`` functions only build the ``pathlib.Path`` directories and the ``mkdir`` are only called in the ``execute_command`` function.